src/Compilers/Core/Portable/SourceGeneration/GeneratorSyntaxWalker.cs

```diff
@@ -2,20 +2,41 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+
 namespace Microsoft.CodeAnalysis
 {
     internal sealed class GeneratorSyntaxWalker : SyntaxWalker
     {
-        private readonly ISyntaxReceiver _syntaxReceiver;
+        private readonly ISyntaxReceiverBase _syntaxReceiver;
+
+        private SemanticModel? _semanticModel;
 
-        internal GeneratorSyntaxWalker(ISyntaxReceiver syntaxReceiver)
+        internal GeneratorSyntaxWalker(ISyntaxReceiverBase syntaxReceiver)
         {
             _syntaxReceiver = syntaxReceiver;
         }
 
+        public void VisitWithModel(SemanticModel model, SyntaxNode node)
+        {
+            Debug.Assert(_semanticModel is null);
+            _semanticModel = model;
```

Comment:
Consider asserting that, if `_syntaxReceiver` is a `ISyntaxReceiverWithContext`, `model` is not null. Or just that `model` is not null at all.